src/Compilers/Core/Portable/NativePdbWriter/ISymUnmanagedWriter.cs

```diff
@@ -73,9 +73,11 @@ internal static class ISymUnmanagedWriter2Helper
     {
         public static unsafe void DefineConstant2(this ISymUnmanagedWriter2 writer, string name, object value, uint sigToken)
         {
-            VariantStructure variant = new VariantStructure();
-            Marshal.GetNativeVariantForObject(value, new IntPtr(&variant));
-            writer.DefineConstant2(name, variant, sigToken);
+            // Xamarin Bug https://bugzilla.xamarin.com/show_bug.cgi?id=37632
```

Comment:
The PDB code should not be commented out. Xamarin should implement the missing Marshal methods, even if they just throw PlatformNotSupportedException.

We have an ```DebugInformationFormat.PortablePdb``` (http://source.roslyn.io/#Microsoft.CodeAnalysis/Emit/DebugInformationFormat.cs) public EmitOption that should be used to emit cross-platform PDB format.

If we don't already we need to do some exception catching and reporting a diagnostic that native PDB format is not supported on non-Windows platforms.